hs_swat_modelinstance/tests/test_swat_modelinstance_metadata.py

```diff
@@ -0,0 +1,824 @@
+        s_params = ["Crop rotation", "Tillage operation"]
+        o_params = "spongebob"
+        self.resSWATModelInstance.metadata.create_element('ModelParameter',
+                                                          model_parameters={s_params[0]: s_params[0],
+                                                                            s_params[1]: s_params[1]},
+                                                          other_parameters=o_params)
+
+        self.resSWATModelInstance.metadata.create_element('ModelInput',
+                                                          warmupPeriodValue='a',
+                                                          rainfallTimeStepType='Daily',
+                                                          rainfallTimeStepValue='c',
+                                                          routingTimeStepType='Daily',
+                                                          routingTimeStepValue='e',
+                                                          simulationTimeStepType='Hourly',
+                                                          simulationTimeStepValue='g',
+                                                          watershedArea='h',
+                                                          numberOfSubbasins='i',
+                                                          numberOfHRUs='j',
+                                                          demResolution='k',
+                                                          demSourceName='l',
+                                                          demSourceURL='m',
+                                                          landUseDataSourceName='n',
+                                                          landUseDataSourceURL='o',
+                                                          soilDataSourceName='p',
+                                                          soilDataSourceURL='q',
+                                                          )
+
+        # there should one content file
+        self.assertEquals(self.resSWATModelInstance.files.all().count(), 1)
+
+        # there should be one format element
+        self.assertEquals(self.resSWATModelInstance.metadata.formats.all().count(), 1)
+
+        # delete content file that we added above
+        hydroshare.delete_resource_file(self.resSWATModelInstance.short_id, self.file_name, self.user)
+        # there should no content file
+        self.assertEquals(self.resSWATModelInstance.files.all().count(), 0)
+
+        # test the core metadata at this point
+        self.assertNotEquals(self.resSWATModelInstance.metadata.title, None)
+
+        # there should be an abstract element
+        self.assertNotEquals(self.resSWATModelInstance.metadata.description, None)
+
+        # there should be one creator element
+        self.assertEquals(self.resSWATModelInstance.metadata.creators.all().count(), 1)
+

```

Comment:
You should be testing here that there are no format elements as  a result of deleting the content file.